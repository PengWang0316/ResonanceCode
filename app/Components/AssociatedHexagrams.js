import React from 'react';
import PropTypes from 'prop-types';

import HexagramImage from './HexagramImage';

const AssociatedHexagrams = ({ hexagram }) => (
  <div>
    <div className="mt-2 font-weight-bold">Associated Hexagrams</div>
    <div className="mt-2 d-flex align-items-center">
      <div>Complementary: #{hexagram.complementary_hexagram_number}&nbsp;&nbsp;</div>
      <div>
        <HexagramImage
          isBlack
          imageNumber={hexagram.complementary_hexagram}
          isFirstImage
          isSmall
        />
      </div>
    </div>
    <div>{hexagram.complementary_hexagram_code}</div>
    <div className="mt-2 d-flex align-items-center">
      <div>Reverse: #{hexagram.reverse_hexagram_number}&nbsp;&nbsp;</div>
      <div>
        <HexagramImage
          isBlack
          imageNumber={hexagram.reverse_hexagram}
          isFirstImage
          isSmall
        />
      </div>
    </div>
    <div>{hexagram.reverse_hexagram_code}</div>
    <div className="mt-2 d-flex align-items-center">
      <div>Hidden Influence: #{hexagram.hidden_influence_hexagram_number}&nbsp;&nbsp;</div>
      <div>
        <HexagramImage
          isBlack
          imageNumber={hexagram.hidden_influence_hexagram}
          isFirstImage
          isSmall
        />
      </div>
    </div>
    <div>{hexagram.hidden_influence_hexagram_code}</div>
  </div>
);
AssociatedHexagrams.propTypes = { hexagram: PropTypes.object.isRequired };
export default AssociatedHexagrams;
