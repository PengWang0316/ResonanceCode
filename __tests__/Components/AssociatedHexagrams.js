import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import AssociatedHexagrams from '../../app/Components/AssociatedHexagrams';

jest.mock('../../app/Components/HexagramImage', () => 'HexagramImage');

describe('AssociatedHexagrams test', () => {
  const defaultProps = {
    hexagram: {
      complementary_hexagram_number: 'complementary_hexagram_number',
      complementary_hexagram: 'complementary_hexagram',
      complementary_hexagram_code: 'complementary_hexagram_code',
      reverse_hexagram_number: 'reverse_hexagram_number',
      reverse_hexagram: 'reverse_hexagram',
      reverse_hexagram_code: 'reverse_hexagram_code',
      hidden_influence_hexagram_number: 'hidden_influence_hexagram_number',
      hidden_influence_hexagram: 'hidden_influence_hexagram',
      hidden_influence_hexagram_code: 'hidden_influence_hexagram_code',
    },
    handleHexagramClick: jest.fn()
  };
  const getShallowComponent = (props = defaultProps) => shallow(<AssociatedHexagrams {...props} />);

  test('AssociatedHexagrams snapshot', () => expect(renderer.create(<AssociatedHexagrams {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('handleHexagramClick click', () => {
    const component = getShallowComponent();
    component.find({ role: 'button' }).at(0).simulate('click');
    expect(defaultProps.handleHexagramClick).toHaveBeenLastCalledWith('complementary_hexagram_number');
    component.find({ role: 'button' }).at(1).simulate('click');
    expect(defaultProps.handleHexagramClick).toHaveBeenLastCalledWith('reverse_hexagram_number');
    component.find({ role: 'button' }).at(2).simulate('click');
    expect(defaultProps.handleHexagramClick).toHaveBeenLastCalledWith('hidden_influence_hexagram_number');
  });
});
