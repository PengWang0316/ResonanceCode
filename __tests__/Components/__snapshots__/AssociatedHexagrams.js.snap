// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssociatedHexagrams test AssociatedHexagrams snapshot 1`] = `
<div>
  <div
    className="mt-2 font-weight-bold"
  >
    Associated Hexagrams
  </div>
  <div
    className="mt-2 d-flex align-items-center"
    onClick={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div>
      Complementary: #
      complementary_hexagram_number
        
    </div>
    <div>
      <HexagramImage
        imageNumber="complementary_hexagram"
        isBlack={true}
        isFirstImage={true}
        isSmall={true}
      />
    </div>
  </div>
  <div>
    complementary_hexagram_code
  </div>
  <div
    className="mt-2 d-flex align-items-center"
    onClick={[Function]}
    role="button"
    tabIndex="-2"
  >
    <div>
      Reverse: #
      reverse_hexagram_number
        
    </div>
    <div>
      <HexagramImage
        imageNumber="reverse_hexagram"
        isBlack={true}
        isFirstImage={true}
        isSmall={true}
      />
    </div>
  </div>
  <div>
    reverse_hexagram_code
  </div>
  <div
    className="mt-2 d-flex align-items-center"
    onClick={[Function]}
    role="button"
    tabIndex="-3"
  >
    <div>
      Hidden Influence: #
      hidden_influence_hexagram_number
        
    </div>
    <div>
      <HexagramImage
        imageNumber="hidden_influence_hexagram"
        isBlack={true}
        isFirstImage={true}
        isSmall={true}
      />
    </div>
  </div>
  <div>
    hidden_influence_hexagram_code
  </div>
</div>
`;
